### PR TITLE
[Driver] Check the environment version except wasm case.

### PR DESCRIPTION
**Pull request by ZijunZhaoCCK**
Add isWasm() check for here: https://github.com/llvm/llvm-project/pull/78655#issuecomment-1928075569

### CODE REVIEW

**Comment by MaskRay** [comment]
Add a test?

**Comment by ZijunZhaoCCK** [comment]
> Add a test?

Done. Add several tests.

**Comment by MaskRay** [comment]
wasm tests in android-* seem weird. A wasm file should be used

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -1443,16 +1443,18 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
...
-  StringRef TripleObjectFormat =
-      Triple.getObjectFormatTypeName(Triple.getObjectFormat());
-  if (Triple.getEnvironmentVersion().empty() && TripleVersionName != "" &&
-      TripleVersionName != TripleObjectFormat) {
-    Diags.Report(diag::err_drv_triple_version_invalid)
-        << TripleVersionName << TC.getTripleString();
-    ContainsError = true;
+  // Check if the environment version is valid except wasm case.
+  if (!TC.getTriple().isWasm()) {
+    llvm::Triple Triple = TC.getTriple();
```
Comment (glandium) [inline clang/lib/Driver/Driver.cpp:1448]: You could do 
```
llvm::Triple Triple = TC.getTriple();
if (!Triple.isWasm()) {
```

File: clang/test/Driver/invalid-version.cpp [hunk 2]
```diff
@@ -14,3 +14,18 @@
 // RUN:   FileCheck --check-prefix=CHECK-TARGET %s
```
Comment (MaskRay) [inline clang/test/Driver/invalid-version.cpp:14]: Sorry, I didn't see the `[clang/test/Driver/invalid-version.cpp](https://github.com/llvm/llvm-project/pull/80783/files#diff-6c20c1f9b36f569382ab3c9daebb660db0eb8666af4beaf5981311b54037c787)` rename. This looks good to me.

File: clang/test/Driver/invalid-version.cpp [hunk 3]
```diff
@@ -14,3 +14,18 @@
...
+
+// RUN: %clang --target=wasm32-unknown-wasi-preview2 -c %s -### 2>&1 | \
+// RUN:   FileCheck --check-prefix=CHECK-TARGET1 %s
+
+// CHECK-TARGET1: "wasm32-unknown-wasi-preview2"
+
+// RUN: %clang --target=wasm32-wasi-pthread -c %s -### 2>&1 | \
+// RUN:   FileCheck --check-prefix=CHECK-TARGET2 %s
+
+// CHECK-TARGET2: "wasm32-unknown-wasi-pthread"
```
Comment (MaskRay) [inline clang/test/Driver/invalid-version.cpp:31]: It's best to check `"-triple"` as well.

**Review by MaskRay — APPROVED** [review]
LGTM with glandium's suggestion.

**Pull request merged**